library.rb

```diff
@@ -0,0 +1,177 @@
+require_relative './objects/author'
+require_relative './objects/book'
+require_relative './objects/client'
+require_relative './services/object_setter'
+require_relative './services/library_helper'
+
+class Library
+  extend ObjectSetter
+  include Language
+  include LibraryHelper
+  attr_reader :books, :authors, :clients, :orders
+
+  def initialize
+    @books = []
+    @authors = []
+    @clients = []
+    @rating = []
+    @orders = []
+  end
+
+  def get_collections(name, &block)
+    file = FileHandler.new(name).parse_file
+    file.each(&block)
+  end
+
+  def parse_books
+    get_collections('books') do |object|
+      entity = Book.new(object['id'],
+                        object['name'],
+                        DateTime.parse(object['written_date']),
+                        DateTime.parse(object['created_at']),
+                        DateTime.parse(object['updated_at']),
+                        object['author_id'].to_i,
+                        object['price'].to_i)
+      @books.push(entity)
+    end
+    @books
+  end
+
+  def parse_authors
+    get_collections('authors') do |object|
+      entity = Author.new(object['first_name'],
+                          object['last_name'],
+                          object['book_id'].to_i)
+      @authors.push(entity)
+    end
+    @authors
+  end
+
+  def parse_clients
+    get_collections('clients') do |object|
+      entity = Client.new(object['id'].to_i,
+                          object['first_name'],
+                          object['last_name'],
+                          object['address'])
+      @clients.push(entity)
+    end
+    @clients
+  end
+
+  def parse_orders
+    get_collections('orders') do |object|
+      entity = Order.new(object['book_id'].to_i,
+                         DateTime.parse(object['created_at']),
+                         object['client_id'].to_i,
+                         object['payed'].to_i)
+      @orders.push(entity)
+    end
+    @orders
+  end
+
+  def load_files
+    @authors = parse_authors
+    @books = parse_books
+    @clients = parse_clients
+    @orders = parse_orders
+  end
+
+  def client_info
+    p phrases_list[:new_client]
+    answer = gets.chomp.downcase
+    until answer == phrases_list[:yes] || answer == phrases_list[:no]
+      p "That's not the anwser we've expected. Let's try once again"
+      answer = gets.chomp
+    end
+    if answer == phrases_list[:yes]
+      p created_client = Client.new_client(clients = @clients)
+      file = FileHandler.new('clients').parse_file
+      file.push(created_client)
+      FileHandler.new('clients').write_file(file)
+      client = Client.new(created_client['id'], created_client['first_name'], created_client['last_name'], created_client['address'])
+    elsif answer == phrases_list[:no]
+      p phrases_list[:enter_details]
+      full_name = gets.chomp
+      client = Client.locate_client(clients = @clients, full_name)
+    end
+    client
+  end
+
+  def write_order(book, client)
```

Comment:
Keep order related methods in Order class

Reply:
Fixed